tests/system/src/test/java/test/com/sun/prism/impl/NGNodeDirtyFlagTest.java

```diff
@@ -0,0 +1,188 @@
+package test.com.sun.prism.impl;
+
+import javafx.application.Application;
+import javafx.geometry.Bounds;
+import javafx.scene.image.WritableImage;
+import javafx.scene.robot.Robot;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javafx.application.Platform;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Line;
+import javafx.stage.Stage;
+import test.util.Util;
+
+import static org.junit.Assert.assertEquals;
+
+public class NGNodeDirtyFlagTest {
+
+    // Used to launch the application before running any test
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Singleton Application instance
+    static MyApp myApp;
+
+    private CountDownLatch latch = new CountDownLatch(1);
+
+    public static class MyApp extends Application {
+
+        private StackPane root;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            root = new StackPane();
+            primaryStage.setScene(new Scene(root, 500, 400));
+            primaryStage.show();
+
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        Util.launch(launchLatch, MyApp.class);
+        assertEquals(0, launchLatch.getCount());
```

Comment:
Util.launch() asserts if launchLatch.countDown() does not occur in 15 seconds, so this line can be removed.

Reply:
Done